eclipse-che/che-server
# Add support for PATs in GitHub Enterprise server

### What does this PR do?
Add necessary checks to the `GithubURLParser` and `AbstractGithubURLParser` classes in order to support PAT for GitHub Enterprise server without configuring oAuth.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

fixes https://github.com/eclipse/che/issues/22582

### How to test this PR?

1. Deploy che with the PR image: `quay.io/eclipse/che-server:pr-603`
2. Create a personal access token in the GitHub Enterprise server instance.
3. Add the token to che through the dashboard user preferences page.
4. Create a workspace from a private repository with a devfile from the GitHub server instance.

see: the workspace starts and the project is cloned.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
